dali/operators/input/video_input.cc

```diff
@@ -0,0 +1,149 @@
+each frame has ``(F, H, W, C)`` layout (where ``F`` is the number of frames in a sequence).
+
+When using ``fn.inputs.video`` operator inside the DALI Pipeline, user needs to provide the data
+using :meth:`Pipeline.feed_input`. When the Operator is fed with data, the Pipeline can be ran
+multiple times and the ``fn.inputs.video`` operator will return consecutive sequences, as long as
+there is enough data remaining. When the source of the frames (the video file) depletes, user needs
+to call another ``feed_input`` to provide the next video file to the operator. This Operator has an
+inner-queue for the data, so the ``feed_input`` may be called multiple times and when given video
+file ends, the Operator will fetch the next one automatically from the top of the queue.
+Running the pipeline while there is no data for the ``fn.inputs.video`` to run results in an error.
+
+This operator takes only one video as and input (i.e. ``input_batch_size=1``) and will return
+batches of sequences, where every output batch will have the size, which has been set during
+the Pipeline object instantiation (``max_batch_size``). When the number of frames in the video
+file does not allow to split the frames uniformly across batches, the last batch returned by
+this operator for a given video will be partial and the last sequence in this batch will be
+determined using ``last_sequence_policy`` parameter. For example::
+
+
+    This is a video that consists of 67 frames (every '-' is a frame):
+    -------------------------------------------------------------------
+
+    User decided that there shall be 5 frames per sequence and the last_sequence_policy='partial':
+    -------------------------------------------------------------------
+    [   ][   ][   ][   ][   ][   ][   ][   ][   ][   ][   ][   ][   ][]
+    -------------------------------------------------------------------
+                      Since there are not enough frames, the last sequence comprises 2 frames.
+
+    The Pipeline has max_batch_size=3, therefore the operator will return batches of 3 sequences:
+    --------------- --------------- --------------- --------------- -------
+    [   ][   ][   ] [   ][   ][   ] [   ][   ][   ] [   ][   ][   ] [   ][]
+    --------------- --------------- --------------- --------------- -------
+                      The last batch is partial and comprises 2 sequences.
+
+
+The difference between ``fn.inputs.video`` and ``fn.readers.video`` is that the former
+reads an encoded video from memory and the latter reads the encoded video from disk.
+
+The difference between ``fn.inputs.video`` and ``fn.decoders.video`` is that the former
+does not decode the whole video file in one go. This behaviour is needed for longer videos. E.g.
+5-min, 4k, 30fps decoded video takes about 1.7 TB of memory.
+
+This operator accepts most of the video containers and file formats. FFmpeg is used to parse
```

Comment:
This statement is too vague. Can we list the supported container and formats instead?

Reply:
I believe it might be problematic, because there are dozens of those.

FTR, I copy-pasted this sentence from `readers.video` documetation: https://docs.nvidia.com/deeplearning/dali/main-user-guide/docs/operations/nvidia.dali.fn.readers.video.html